### ISSUE TITLE
[Android] Cannot play video when navigating away and back from the video page

### ISSUE THREAD

**Issue by aiviet** [martijn00/XamarinMediaManager#239]
**To reproduce the bug:**
1. On the video page, play some video
2. Navigate to a new page
3. Navigate back to the video page
4. Video will not play from this point onward 

**Reason:**
On step (1), video plays no problem because isPlayerReady is false and hence "Init()" is excuted
```

public async Task Play(IMediaFile mediaFile = null)
{
	if (VideoViewCanvas == null)
		throw new System.Exception("No canvas set for video to play in");

	if (isPlayerReady == false)
	{
		//await Task.Delay(100);
		Init();
		isPlayerReady = true;
	}
	...
}
```

On step (2), navigating away from the video page will trigger the dispose of VideoViewCanvas

On step (4), new VideoViewCanvas is created but "Init()" is not excuted because isPlayerReady is already set to true from the previous play in step 1.
Without Init, OnPrepared event handler is not attached to the new VideoViewCanvas and in turn no video renedering because VideoViewCanvas.Start() will not be excuted.

```
public void OnPrepared(MediaPlayer mp)
{
	if(Status == MediaPlayerStatus.Buffering)
		VideoViewCanvas.Start();

	Status = MediaPlayerStatus.Playing;
}

```
**Walk-around:**
Add 2 more steps below:
5. Call the Pause method to set the Status = MediaPlayerStatus.Paused
6. Call the Play method again. With Status == MediaPlayerStatus.Paused, VideoViewCanvas.Start() will be excuted

**Comment by Apfelcheck** [comment]
Hi,

i have the same problem. I do not understood the fix exactly. The fix should be done in the source of the framework or can I do anything to get my app running also with the second video playing?

If the fix has to be done in the source them when Martijn will be so glad and will publish a fixed version in the NuGet? Or have I integrate this project in my project too?

Regards and thanks to all

Waldemar

**Comment by aiviet** [comment]
@Apfelcheck I already created a pull request that fixed #238 and #239 here: https://github.com/martijn00/XamarinMediaManager/pull/240

If you cannot wait for the fix above to be published, you can do this in your app as a "brute force" walk-around solution:

```
protected override void OnAppearing()
{	        
	base.OnAppearing();
	
	//To get around issue #239
	PlaybackController.Pause();
	PlaybackController.Play();
}
		
void PlayClicked(object sender, System.EventArgs e)
{            	
	PlaybackController.Play();
}
```

**Comment by Apfelcheck** [comment]
Thank you for the help. Do you know, if it is possible to use the MediaManger in a ContentView only? 
Regards

Waldemar

**Comment by Apfelcheck** [comment]
The solution mentioned above is not working.

**Comment by Apfelcheck** [comment]
@martijn00 👍 but, when will you publish to NuGet please?

**Comment by aiviet** [comment]
@Apfelcheck I am using MediaManager in ContentView no problem.

Maybe your code has a different problem than the walk around's.

**Comment by Apfelcheck** [comment]
Thank you for the answers. My code is very simple. I reallly don´t know what the problem is. Can you contact me by PN? Using Facebook or Google Plus?

**Comment by aiviet** [comment]
@Apfelcheck You can contact me via my profile email.

**Comment by raaajeeev** [comment]
Follow the below along with the previous suggestions in the thread by @aiviet and this issue will be resolved.

**Android:
You must request AccessWifiState, Internet, MediaContentControl and WakeLock permissions**

**Comment by aiviet** [comment]
Fixed in #240
